### PR TITLE
Fix links to Iris documentation

### PR DESCRIPTION
**Pull request by jvegasbsc**
Iris documentation has been moved to ReadTheDocs. This PR updates our links to it

***

## Before you get started

-   [ ] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## Checklist

-   [x] PR has a descriptive title for the [changelog](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#branches-pull-requests-and-code-review)
-   [x] Code follows the [style guide](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#code-style)
-   [ ] [Documentation](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#documentation) is available for new functionality
-   [x] YAML files pass [`pre-commit`](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#pre-commit) or [`yamllint`](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/community/introduction.html#yaml) checks
-   [x] [Circle/CI tests pass](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#branches-pull-requests-and-code-review)
-   [x] [Codacy code quality checks pass](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#branches-pull-requests-and-code-review)
-   [x] [Documentation builds successfully](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#branches-pull-requests-and-code-review) on readthedocs
-   [x] [Unit tests](https://docs.esmvaltool.org/projects/esmvalcore/projects/esmvalcore/en/latest/contributing.html#contributing-to-the-esmvalcore-package) are available

If you make backwards incompatible changes to the recipe format:

-   [ ] Update [ESMValTool](https://github.com/esmvalgroup/esmvaltool) and link the pull request(s) in the description

### CODE REVIEW

File: doc/recipe/preprocessor.rst [hunk 1]
```diff
@@ -606,11 +606,11 @@ The schemes used for the interpolation and extrapolation operations needed by
...
-* ``linear``: `Linear(extrapolation_mode='mask') <https://scitools.org.uk/iris/docs/latest/iris/iris/analysis.html#iris.analysis.Linear>`_.
-* ``linear_extrapolate``: `Linear(extrapolation_mode='extrapolate') <https://scitools.org.uk/iris/docs/latest/iris/iris/analysis.html#iris.analysis.Linear>`_.
-* ``nearest``: `Nearest(extrapolation_mode='mask') <https://scitools.org.uk/iris/docs/latest/iris/iris/analysis.html#iris.analysis.Nearest>`_.
-* ``area_weighted``: `AreaWeighted() <https://scitools.org.uk/iris/docs/latest/iris/iris/analysis.html#iris.analysis.AreaWeighted>`_.
-* ``unstructured_nearest``: `UnstructuredNearest() <https://scitools.org.uk/iris/docs/latest/iris/iris/analysis.html#iris.analysis.UnstructuredNearest>`_.
+* ``linear``: `Linear(extrapolation_mode='mask') <https://scitools-iris.readthedocs.io/en/latest/iris/iris/analysis.html#iris.analysis.Linear>`_.
+* ``linear_extrapolate``: `Linear(extrapolation_mode='extrapolate') <https://scitools-iris.readthedocs.io/en/latest/iris/iris/analysis.html#iris.analysis.Linear>`_.
+* ``nearest``: `Nearest(extrapolation_mode='mask') <https://scitools-iris.readthedocs.io/en/latest/iris/iris/analysis.html#iris.analysis.Nearest>`_.
```
Comment (stefsmeets) [inline doc/recipe/preprocessor.rst:611]:
```suggestion
* ``nearest``: `Nearest(extrapolation_mode='mask') <https://scitools-iris.readthedocs.io/en/latest/generated/api/irisanalysis.html#iris.analysis.Nearest>`_.
```

**Review by stefsmeets — DISMISSED** [review]
Nice work, but there are still some broken links. I think I got all of them in my comments 👍

File: doc/recipe/preprocessor.rst [hunk 2]
```diff
@@ -1098,9 +1098,9 @@ See also :func:`esmvalcore.preprocessor.regrid_time`.
 This function allows the user to apply a filter to the timeseries data. This filter may be
 of the user's choice (currently only the ``low-pass`` Lanczos filter is implemented); the
 implementation is inspired by this `iris example
-<https://scitools.org.uk/iris/docs/latest/examples/General/
+<https://scitools-iris.readthedocs.io/en/latest/examples/General/
```
[the remark this answers is not in the document]
Comment (jvegasbsc) [reply doc/recipe/preprocessor.rst]: Fixed now!

**Comment by jvegasbsc** [comment]
It looks like I checked the few ones that just needed a replacement in the webpage name 🤦 

Thanks @stefsmeets

File: doc/recipe/preprocessor.rst [hunk 3]
```diff
@@ -65,7 +65,7 @@ the preprocessor structure see the :ref:`Preprocessors`.
 
 Each of the preprocessor operations is written in a dedicated python module and
 all of them receive and return an Iris `cube
-<https://scitools.org.uk/iris/docs/v2.0/iris/iris/cube.html>`_ , working
+<https://scitools-iris.readthedocs.io/en/latest/generated/api/iris/cube.html>`_ , working
```
Comment (bouweandela) [inline doc/recipe/preprocessor.rst:68]: You can just use interphinx to directly link to iris functions and classes in most places, that avoids this problem recurring if the documentation is moved again. E.g. this link can be replace by
```
:obj:`iris.cube.Cube`
```

Comment (jvegasbsc) [reply doc/recipe/preprocessor.rst:68]: Great to know! Thanks, Bouwe!

**Comment by jvegasbsc** [comment]
Everything should be fixed now

File: doc/recipe/preprocessor.rst [hunk 4]
```diff
@@ -64,8 +64,8 @@ reducing the data processing load on the diagnostics side.  For an overview of
 the preprocessor structure see the :ref:`Preprocessors`.
 
 Each of the preprocessor operations is written in a dedicated python module and
-all of them receive and return an Iris `cube
-<https://scitools.org.uk/iris/docs/v2.0/iris/iris/cube.html>`_ , working
+all of them receive and return an Iris cube
+:obj:`iris.cube.Cube`, working
```
Comment (stefsmeets) [inline doc/recipe/preprocessor.rst:68]:
```suggestion
all of them receive and return an instance of
:obj:`iris.cube.Cube`, working
```

File: doc/recipe/preprocessor.rst [hunk 5]
```diff
@@ -537,8 +537,8 @@ engine uses interpolation and extrapolation, with various schemes). The primary
 difference is that interpolation is based on sample data points, while
 regridding is based on the horizontal grid of another cube (the reference grid).
 
-The underlying regridding mechanism in ESMValTool uses the `cube.regrid()
-<https://scitools.org.uk/iris/docs/latest/iris/iris/cube.html#iris.cube.Cube.regrid>`_
+The underlying regridding mechanism in ESMValTool uses the cube.regrid()
+:obj:`iris.cube.Cube.regrid`
```
Comment (stefsmeets) [inline doc/recipe/preprocessor.rst:541]:
```suggestion
The underlying regridding mechanism in ESMValTool uses
:obj:`iris.cube.Cube.regrid`
```

File: doc/recipe/preprocessor.rst [hunk 6]
```diff
@@ -606,11 +606,11 @@ The schemes used for the interpolation and extrapolation operations needed by
...
-* ``linear``: `Linear(extrapolation_mode='mask') <https://scitools.org.uk/iris/docs/latest/iris/iris/analysis.html#iris.analysis.Linear>`_.
-* ``linear_extrapolate``: `Linear(extrapolation_mode='extrapolate') <https://scitools.org.uk/iris/docs/latest/iris/iris/analysis.html#iris.analysis.Linear>`_.
-* ``nearest``: `Nearest(extrapolation_mode='mask') <https://scitools.org.uk/iris/docs/latest/iris/iris/analysis.html#iris.analysis.Nearest>`_.
-* ``area_weighted``: `AreaWeighted() <https://scitools.org.uk/iris/docs/latest/iris/iris/analysis.html#iris.analysis.AreaWeighted>`_.
-* ``unstructured_nearest``: `UnstructuredNearest() <https://scitools.org.uk/iris/docs/latest/iris/iris/analysis.html#iris.analysis.UnstructuredNearest>`_.
+* ``linear``: Linear(extrapolation_mode='mask'), see :obj:`iris.analysis.Linear`.
+* ``linear_extrapolate``: Linear(extrapolation_mode='extrapolate'), see :obj:`iris.analysis.Linear`.
+* ``nearest``: Nearest(extrapolation_mode='mask'), see :obj:`iris.analysis.Nearest`.
+* ``area_weighted``: AreaWeighted() :obj:`iris.analysis.AreaWeighted`.
+* ``unstructured_nearest``: UnstructuredNearest(), see :obj:`iris.analysis.UnstructuredNearest`.
```
Comment (stefsmeets) [inline doc/recipe/preprocessor.rst:613]:
```suggestion
* ``linear``: Linear(extrapolation_mode='mask'), see :obj:`iris.analysis.Linear`.
* ``linear_extrapolate``: ``Linear(extrapolation_mode='extrapolate')``, see :obj:`iris.analysis.Linear`.
* ``nearest``: ``Nearest(extrapolation_mode='mask')``, see :obj:`iris.analysis.Nearest`.
* ``area_weighted``: ``AreaWeighted()``, see :obj:`iris.analysis.AreaWeighted`.
* ``unstructured_nearest``: ``UnstructuredNearest()``, see :obj:`iris.analysis.UnstructuredNearest`.
```

File: doc/recipe/preprocessor.rst [hunk 7]
```diff
@@ -1098,9 +1098,8 @@ See also :func:`esmvalcore.preprocessor.regrid_time`.
 This function allows the user to apply a filter to the timeseries data. This filter may be
 of the user's choice (currently only the ``low-pass`` Lanczos filter is implemented); the
 implementation is inspired by this `iris example
-<https://scitools.org.uk/iris/docs/latest/examples/General/
-SOI_filtering.html?highlight=running%20mean>`_ and uses aggregation via a
-`rolling window <https://scitools.org.uk/iris/docs/v2.0/iris/iris/cube.html#iris.cube.Cube.rolling_window>`_.
+<https://scitools-iris.readthedocs.io/en/latest/generated/gallery/general/plot_SOI_filtering.html>`_ and uses aggregation via a
+rolling window obj:`iris.cube.Cube.rolling_window`.
```
Comment (stefsmeets) [inline doc/recipe/preprocessor.rst:1102]:
```suggestion
<https://scitools-iris.readthedocs.io/en/latest/generated/gallery/general/plot_SOI_filtering.html>`_ and uses aggregation via :obj:`iris.cube.Cube.rolling_window`.
```

File: doc/recipe/preprocessor.rst [hunk 8]
```diff
@@ -1412,8 +1411,8 @@ This function extracts the peak-to-peak amplitude (maximum value minus minimum
 value) of a field aggregated over specified coordinates. Its only argument is
 ``coords``, which can either be a single coordinate (given as :obj:`str`) or
 multiple coordinates (given as :obj:`list` of :obj:`str`). Usually, these
-coordinates refer to temporal `categorised coordinates
-<https://scitools.org.uk/iris/docs/latest/iris/iris/coord_categorisation.html>`_
+coordinates refer to temporal categorised coordinates
+obj:`iris.coord_categorisation`
```
Comment (stefsmeets) [inline doc/recipe/preprocessor.rst:1415]:
```suggestion
coordinates refer to temporal categorised coordinates
:obj:`iris.coord_categorisation`
```

File: esmvalcore/preprocessor/_time.py [hunk 9]
```diff
@@ -778,16 +777,13 @@ def timeseries_filter(cube,
...
+    <https://scitools-iris.readthedocs.io/en/latest/generated/gallery/general/plot_SOI_filtering.html?highlight=running%20mean>`_
 
     Apply each filter using the rolling_window method used with the weights
     keyword argument. A weighted sum is required because the magnitude of
     the weights are just as important as their relative sizes.
 
-    See also the `iris rolling window
-    <https://scitools.org.uk/iris/docs/v2.0/iris/iris/
-    cube.html#iris.cube.Cube.rolling_window>`_
+    See also the iris rolling window obj:`iris.cube.Cube.rolling_window`.
```
Comment (stefsmeets) [inline esmvalcore/preprocessor/_time.py:786]:
```suggestion
    See also the iris rolling window :obj:`iris.cube.Cube.rolling_window`.
```

**Review by stefsmeets — COMMENTED** [review]
Nice work! Most links I clicked were OK, it just seems a few items are missing colons. I also updated the text in a few places to make it a bit nicer to read.

**Comment by jvegasbsc** [comment]
Suggestions applied!

**Review by nielsdrost — APPROVED** [review]
Doubled checked link, nice work @jvegasbsc and thanks for the review @stefsmeets!

**Pull request merged**